app/shared/favorite-button/FavoriteButton.spec.js

```diff
@@ -0,0 +1,52 @@
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
+import simple from 'simple-mock';
+
+import FavoriteButton from './FavoriteButton';
+
+describe('shared/favorite-button/FavoriteButton', () => {
+  const defaultProps = {
+    favorited: true,
+    onClick: simple.mock(),
+  };
+
+  function getWrapper(props) {
+    return shallow(<FavoriteButton {...defaultProps} {...props} />);
+  }
+
+  it('is a button', () => {
+    expect(getWrapper().is('button')).to.be.true;
+  });
+
+  it('has favorite-button class name', () => {
+    expect(getWrapper().prop('className')).to.equal('favorite-button');
+  });
+
+  it('passes onClick prop', () => {
+    expect(getWrapper().prop('onClick')).to.deep.equal(defaultProps.onClick);
+  });
+
+  describe('favorited', () => {
+    it('has correct text', () => {
+      expect(getWrapper().find('.favorite-text').text()).to.equal('Poista suosikeista');
+    });
+    it('has a heart glyphicon if favorited', () => {
```

Comment:
- [ ] The "if favorited" part in the description is not necessary since it's already in the describe.